### ISSUE TITLE
[java] AvoidLosingExceptionInformation does not trigger when inside if-else

### ISSUE THREAD

**Issue by csrma** [pmd/pmd#5881]
**Affects PMD Version:** 7.15.0

**Rule:** AvoidLosingExceptionInformation

Please provide the rule name and a link to the rule documentation: [AvoidLosingExceptionInformation](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#avoidlosingexceptioninformation)

**Description:** The code block below is kind of straight forward. In both branches, it violates the rule. But it is not detected. Likely because the XPath expression only checks for `ExpressionStatement` *directly* under `CatchClause/Block`. It does not consider any other intermediate nodes that might be present.

**Code Sample demonstrating the issue:**

```java
class AvoidLosingExceptionInformation {
    public void showBug() {
        try {
            throw new Exception("This is a test exception");
        } catch (Exception se) {
            if (true) {
                se.getMessage();    // should raise flag
            } else {
                se.getLocalizedMessage();    // should raise flag
            }
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 7, 9, but doesn't. This is a false-negative.

**Running PMD through:** *[CLI]*
